packages/android_alarm_manager_plus/android/src/main/java/dev/fluttercommunity/plus/androidalarmmanager/FlutterBackgroundExecutor.java

```diff
@@ -193,7 +195,15 @@ public void executeDartCallbackInBackgroundIsolate(Intent intent, final CountDow
     // attention to the type of the callback handle as storing this value in a
     // variable of the wrong size will cause the callback lookup to fail.
     long callbackHandle = intent.getLongExtra("callbackHandle", 0);
-
+    String paramsJsonString = intent.getStringExtra("params");
+    JSONObject params = null;
+    if (!TextUtils.isEmpty(paramsJsonString)) {
+      try {
+        params = new JSONObject(paramsJsonString);
+      } catch (JSONException e) {
+        e.printStackTrace();
```

Comment:
Wouldn't it be better to throw the exception up? We want users to know if something goes wrong, rather than fail silently,

Reply:
I think this exception can not be thrown. Because the params must convert to json first to passing through channel . Although I think it can be happening , but I rethrow the exception as IllegalArgumentException.